### PR TITLE
Feature/export

### PR DESCRIPTION
**Pull request by JamesSCBD**

### CODE REVIEW

File: components/ExportModal.vue [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+          const subkeys = getRecordKeys(obj[key][i]);
+          keys          = keys.concat(subkeys.map((subkey) => `${key}[${i}].${subkey}`));
+        }
+      }
+    } else {
+      keys.push(key);
+    }
+  }
+  return keys;
+}
```
Comment (sbilodeau) [inline components/ExportModal.vue:160]: Do you think it's possible simplify this block of code?
Make it "readable"

File: components/ExportModal.vue [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+function getRecordKeys(obj) {
+  let keys = [];
+  for (const key in obj) { // eslint-disable-line
+    if (typeof obj[key] === 'object' && !Array.isArray(obj[key])) {
+      const subkeys = getRecordKeys(obj[key]);
+      keys          = keys.concat(subkeys.map((subkey) => `${key}.${subkey}`));
+    } else if (Array.isArray(obj[key])) {
+      for (let i = 0; i < obj[key].length; i++) {
+        if (typeof (obj[key][i]) === 'string') {
+          keys.push(`${key}[${i}]`);
```
Comment (sbilodeau) [inline components/ExportModal.vue:149]: Do you need special handling for boolean. date times, numbers etc?

File: components/list/ContactsList.vue [hunk 3]
```diff
@@ -129,6 +160,19 @@ export default {
...
+    totalRecords() { return this.totalRows ? `${this.totalRows} records` : 'No records'; },
+  },
+  watch: {
+    async baseQuery() {
+      const query = this.baseQuery || {};
+      if (_.isEmpty(query)) { this.selectedContactResult = {}; }
+
+      const result               = await this.$kronosApi.queryContacts({ ...query, limit: 10 });
+      this.selectedContactResult = result;
+    },
```
Comment (sbilodeau) [inline components/list/ContactsList.vue:175]: Don't do that in a `watch`, you are making 2 api calls every time (one for the search and one to create a result objet for the task. 

Create the result object and selectedContactResult ONLY when the user click the task using a method on the @click handler here  https://github.com/scbd/kronos-web/pull/46/files#diff-5ca074a1bac26b7d55b64ccc84db9f8bR17

File: components/ExportModal.vue (hunk 2 above)
Comment (JamesSCBD) [reply components/ExportModal.vue]: In current data no need to add if data comes like: 
```
mobile: [ 1234, 233],
dates: [12-06-2020, 12-06-2020] ,
boolean: [true, false]
```
Then we need to add them

File: components/ExportModal.vue (hunk 1 above)
Comment (JamesSCBD) [reply components/ExportModal.vue]: i am working on it

File: components/list/ContactsList.vue (hunk 3 above)
Comment (JamesSCBD) [reply components/list/ContactsList.vue]: Done!

**Comment by JamesSCBD** [comment]
> This is working well. Please add organization

Done!

**Pull request merged**